File: src/main/java/io/github/takenoha/towerdefense/persistence/TeamResourceSettlement.java
package io.github.takenoha.towerdefense.persistence;

import io.github.takenoha.towerdefense.domain.DefensePhase;
import java.util.Objects;
import java.util.UUID;

/** Immutable terminal totals used for the player-facing settlement message. */
public record TeamResourceSettlement(
        UUID eventId,
        UUID teamId,
        DefensePhase phase,
        long defensePoints,
        long enhancementPoints) {
    public TeamResourceSettlement {
        Objects.requireNonNull(eventId, "eventId");
        Objects.requireNonNull(teamId, "teamId");
        Objects.requireNonNull(phase, "phase");
        if (defensePoints < 0L || enhancementPoints < 0L) {
            throw new IllegalArgumentException("settlement quantities must not be negative");
        }
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/TeamResourceSettlement.kt
package io.github.takenoha.towerdefense.persistence

import io.github.takenoha.towerdefense.domain.DefensePhase
import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** Immutable terminal totals used for the player-facing settlement message. */
@JvmRecord
data class TeamResourceSettlement(
    val eventId: UUID,
    val teamId: UUID,
    val phase: DefensePhase,
    val defensePoints: Long,
    val enhancementPoints: Long,
) {
    init {
        Objects.requireNonNull(eventId, "eventId")
        Objects.requireNonNull(teamId, "teamId")
        Objects.requireNonNull(phase, "phase")
        if (defensePoints < 0L || enhancementPoints < 0L) {
            throw IllegalArgumentException("settlement quantities must not be negative")
        }
    }
}
